File: app/src/main/java/com/faendir/lightning_launcher/multitool/util/provider/QueryDataSource.java
package com.faendir.lightning_launcher.multitool.util.provider;

import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * @author lukas
 * @since 10.08.18
 */
public interface QueryDataSource extends DataSource {
    @Nullable
    Cursor query(@NonNull Context context, @NonNull Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/util/provider/QueryDataSource.kt
package com.faendir.lightning_launcher.multitool.util.provider

import android.content.Context
import android.database.Cursor
import android.net.Uri

/**
 * @author lukas
 * @since 10.08.18
 */
interface QueryDataSource : DataSource {
    fun query(context: Context, uri: Uri, projection: Array<String>?, selection: String?, selectionArgs: Array<String>?, sortOrder: String?): Cursor?
}
